### PR TITLE
fix(events): events for all kinds of searches

### LINKED ISSUE #513: Events for all kind of new search
In 2.20 we added searchQueueEmpty which is great to figure out if there is any search pending. This takes into account all the searches. However we can't get the info about searches for `searchForFacetValues`, `searchOnce`.

We should add events when a search is triggered by those methods with the name of the method as the name of the event:
 - `searchForFacetValues`
 - `searchOnce`

In a later version, we could introduce scoped events but that would be a breaking change as search is already used.

Even though it sounds like a feature, it's actually something that prevents us from implementing a correct loading indicator, which is the original purpose of the new event >> bug.

### PR DESCRIPTION
**Pull request by bobylito**
fix #513

### CODE REVIEW

File: documentation-src/metalsmith/content/reference.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ during the search cycle with events:
  - `result`: when the results are retrieved from Algolia
  - `error`: when Algolia sends back an error
  - `searchQueueEmpty`: when there is no more on going searches
+ - `searchOnce`: when a search is sent to Algolia using `searchForFacetValues`
```
Comment (Haroenv) [inline documentation-src/metalsmith/content/reference.md]: this is the other way around with L17

File: src/algoliasearch.helper.js [hunk 2]
```diff
@@ -180,6 +202,8 @@ AlgoliaSearchHelper.prototype.searchOnce = function(options, cb) {
 
   this._currentNbQueries++;
 
+  this.emit('searchOnce');
```
Comment (Haroenv) [inline src/algoliasearch.helper.js]: Does this not need to be tested?

Comment (bobylito) [inline src/algoliasearch.helper.js]: Good question :) I guess we could test that! :)

File: documentation-src/metalsmith/content/reference.md (hunk 1 above)
Comment (bobylito) [inline documentation-src/metalsmith/content/reference.md]: arg!

**Comment by vvo** [comment]
Should we also add the corresponding search states like the search event does? Does it makes sense?

**Comment by bobylito** [comment]
> Should we also add the corresponding search states like the search event does? Does it makes sense?

What do you mean?

File: src/algoliasearch.helper.js (hunk 2 above)
Comment (bobylito) [inline src/algoliasearch.helper.js]: Now it is tested 👍

Comment (Haroenv) [inline src/algoliasearch.helper.js]: Don't forget to enjoy your holiday @bobylito

File: documentation-src/metalsmith/content/reference.md [hunk 3]
```diff
@@ -13,6 +13,8 @@ during the search cycle with events:
  - `result`: when the results are retrieved from Algolia
  - `error`: when Algolia sends back an error
  - `searchQueueEmpty`: when there is no more on going searches
```
Comment (Haroenv) [inline documentation-src/metalsmith/content/reference.md]: I'm seeing this now, while it was made earlier, but this should be `ongoing` or `pending`

Comment (bobylito) [inline documentation-src/metalsmith/content/reference.md]: For me it's the same, but please which is which? #lostInTranslation

Comment (Haroenv) [inline documentation-src/metalsmith/content/reference.md]: ongoing means that there are searches that are busy, while pending means that there's something to happen with them still. Both almost the same

**Comment by bobylito** [comment]
@vvo I added the data to the events

**Comment by vvo** [comment]
LGTM

**Pull request merged**